mm2src/coins/utxo/utxo_common.rs

```diff
@@ -923,6 +922,20 @@ fn pubkey_from_script_sig(script: &Script) -> Result<H264, String> {
     Ok(pubkey.serialize_compressed().into())
 }
 
+/// Extracts pubkey from witness script
+fn pubkey_from_witness_script(witness_script: &[Bytes]) -> Result<H264, String> {
+    if witness_script.len() != 2 {
+        return ERR!("Invalid witness length {}", witness_script.len());
+    }
+
+    let signature = witness_script[0].clone().take();
+    try_s!(Signature::parse_der(&signature[..signature.len() - 1]));
```

Comment:
I think it's better to check `signature.is_empty()`

Reply:
this checks if a signature is valid like https://github.com/KomodoPlatform/atomicDEX-API/blob/bf24916c332c6563fb72f52fea917668c3b61416/mm2src/coins/utxo/utxo_common.rs#L898
I don't get how `signature.is_empty()` would work.

Reply:
If `signature` is empty, this will attempt to subtract with overflow range `[0..-1]`

Reply:
ooh, I thought you meant to replace the code with `signature.is_empty()`. The check should be added you are right :)

Reply:
Done